cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/CreateSchedulesStage.java

```diff
@@ -121,13 +123,15 @@ private void createSchedule(ProgramId programId, ScheduleSpecification scheduleS
 
   private void addProgramSchedule(ProgramId programId, ScheduleSpecification scheduleSpec) {
     Schedule schedule = scheduleSpec.getSchedule();
-    if (schedule instanceof TimeSchedule) {
+    if (schedule instanceof TimeSchedule || schedule instanceof StreamSizeSchedule) {
       ProgramSchedule programSchedule =
-        Schedulers.toProgramSchedule((TimeSchedule) schedule, programId, scheduleSpec.getProperties());
+        Schedulers.toProgramSchedule(schedule, programId, scheduleSpec.getProperties());
       try {
         programScheduler.addSchedule(programSchedule);
       } catch (AlreadyExistsException e) {
-        LOG.warn("ProgramSchedule {} already exists for program {}", schedule.getName(), programId);
+        LOG.warn("Schedule {} already exists for program {}", schedule.getName(), programId);
+      } catch (BadRequestException e) {
+        LOG.warn("Schedule {} for program {} is not valid: {}", schedule.getName(), programId, e.getMessage());
```

Comment:
Not sure, I only added the BadRequest exception. This should go away with the refactoring, though.